catalog/app/containers/Bucket/PackageCopyDialog.js

```diff
@@ -0,0 +1,435 @@
+import * as R from 'ramda'
+import { FORM_ERROR } from 'final-form'
+import * as React from 'react'
+import * as RF from 'react-final-form'
+import * as M from '@material-ui/core'
+
+import AsyncResult from 'utils/AsyncResult'
+import * as APIConnector from 'utils/APIConnector'
+import * as AWS from 'utils/AWS'
+import * as Data from 'utils/Data'
+import Delay from 'utils/Delay'
+import * as NamedRoutes from 'utils/NamedRoutes'
+import StyledLink from 'utils/StyledLink'
+import tagged from 'utils/tagged'
+import * as validators from 'utils/validators'
+
+import * as PD from './PackageDialog'
+import * as requests from './requests'
+
+async function requestPackageCopy(
+  req,
+  {
+    commitMessage,
+    copyData,
+    hash,
+    initialName,
+    meta,
+    name,
+    sourceBucket,
+    targetBucket,
+    workflow,
+  },
+) {
+  try {
+    return req({
+      endpoint: '/packages/promote',
+      method: 'POST',
+      body: {
+        copy_data: copyData,
+        message: commitMessage,
+        meta: PD.getMetaValue(meta),
+        name,
+        parent: {
+          top_hash: hash,
+          registry: `s3://${sourceBucket}`,
+          name: initialName,
+        },
+        registry: `s3://${targetBucket}`,
+        workflow: PD.getWorkflowApiParam(workflow.slug),
+      },
+    })
+  } catch (e) {
+    return { [FORM_ERROR]: e.message || PD.ERROR_MESSAGES.MANIFEST }
+  }
+}
+
+const useCopyDataSwitcherStyles = M.makeStyles((t) => ({
+  root: {
+    marginTop: t.spacing(2),
+  },
+  icon: {
+    color: t.palette.text.hint,
+    marginLeft: t.spacing(1),
+    verticalAlign: '-4px',
+  },
+}))
+
+function CopyDataSwitcher({ input: { onChange, value }, targetBucket, sourceBucket }) {
+  const classes = useCopyDataSwitcherStyles()
+
+  const handleChange = React.useCallback((event) => onChange(event.target.checked), [
+    onChange,
+  ])
+
+  const hint = (
+    <span>
+      Copy data
+      <M.Tooltip
+        title={
+          value
+            ? `Files will be copied from "${sourceBucket}" to "${targetBucket}"`
```

Comment:
this may be incorrect for some cases, because files may be in different bucket(s), not necessarily in sourceBucket. @akarve pls wordsmith
```suggestion
            ? `Files will be copied to s3://${targetBucket}`
```